packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/entityFeatures.ts

```diff
@@ -191,6 +196,216 @@ function cacheGetNeighborEntityElement(
+    keys: [Keys.LEFT],
+    shouldHandleEvent: (event: PluginKeyboardEvent, editor: IEditor) => {
+        return getIsDelimiterAtCursor(event, editor, false /* checkBefore */);
+    },
+    handleEvent(event: PluginKeyboardEvent, editor: IEditor) {
+        const delimiter = cacheDelimiter(event, false /* checkBefore */);
+        if (!delimiter) {
+            return;
+        }
+        const delimiterBefore = delimiter.previousElementSibling?.previousElementSibling;
+
+        if (delimiterBefore) {
+            const selection = delimiterBefore.ownerDocument.getSelection();
+
+            if (selection) {
+                event.rawEvent.preventDefault();
+                editor.runAsync(() => {
+                    const position = new Position(delimiterBefore, PositionType.Before);
+                    if (event.rawEvent.shiftKey) {
+                        selection.extend(position.element, position.offset);
+                    } else {
+                        selection.setPosition(position.element, position.offset);
+                    }
+                });
+            }
+        }
+    },
+};
+
+/**
+ * Content edit feature to move the cursor from Delimiter Before Entity to the delimiter After
+ */
+const MoveAfterDelimiterFeature: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.RIGHT],
+    shouldHandleEvent: (event: PluginKeyboardEvent, editor: IEditor) => {
+        return getIsDelimiterAtCursor(event, editor, true /* checkBefore */);
+    },
+    handleEvent(event: PluginKeyboardEvent, editor: IEditor) {
+        const delimiter = cacheDelimiter(event, true /* checkBefore */);
+        if (!delimiter) {
+            return;
+        }
+        const delimiterAfter = delimiter.nextElementSibling?.nextElementSibling;
+
+        if (delimiterAfter) {
+            event.rawEvent.preventDefault();
+            editor.runAsync(() => {
+                let offset = 0;
+                delimiterAfter.childNodes.forEach((node, index) => {
+                    if (node.textContent === ZERO_WIDTH_SPACE) {
+                        offset = index;
+                    }
+                });
+                if (event.rawEvent.shiftKey) {
+                    const selection = delimiterAfter.ownerDocument.getSelection();
+                    selection?.extend(delimiterAfter, offset + 1);
+                } else {
+                    editor.select(delimiterAfter, offset + 1);
+                }
+            });
+        }
+    },
+};
+
+/**
+ * Content edit Feature to trigger a Delete Entity Operation when one of the Delimiter is about to be removed with DELETE or Backspace
+ */
+const RemoveEntityBetweenDelimitersFeature: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.BACKSPACE, Keys.DELETE],
+    shouldHandleEvent(event: PluginKeyboardEvent, editor: IEditor) {
+        const range = editor.getSelectionRange();
+        if (!range.collapsed) {
```

Comment:
If press SHIFT+Left/Right, I think we should also handle the selection. For example, cursor is after delimiter, press SHIFT+LEFT, need to select the whole entity as well as two delimiters.

Reply:
Added in the latest commit:
![Delimiter3](https://user-images.githubusercontent.com/8291124/221710494-e72c9eb5-21e1-4d54-aa4e-7eb55db63656.gif)